### ISSUE TITLE
Implement long TimeZone name for TimeZoneFormat

### ISSUE THREAD

**Issue by nordzilla** [unicode-org/icu4x#986]
We still need to implement the `vv` pattern for TimeZoneFormat, which his the long time zone name. This is the canonical IANA time zone identifier.

Depends on:
* #992 (TzdbDataProvider)

**Comment by sffc** [comment]
Yeah, either

1. Add another associated type for this to TimeZoneModel (don't love this)
2. Add the reverse mapper to the formatter (don't live this either)

**Comment by robertbastian** [comment]
Adding the reverse mapper would not round-trip, as there are more IANA IDs than CLDR IDs. Is round-tripping a requirement?

**Comment by sffc** [comment]
That's correct; we'd display the CLDR IANA name ("Europe/Kyiv" even if the input was "Europe/Kiev")

**Comment by sffc** [comment]
The field is `VV`, not `vv`

It appears we handle `VVVV`, `VVV`, and `V`, but not `VV`? Here's where I'm looking:

https://github.com/unicode-org/icu4x/blob/f16ea510216a4f5a1f184c6db4aaed2e6ed60d8c/components/datetime/src/format/datetime.rs#L483

Need to think about data loading here.

**Comment by sffc** [comment]
Since this requires adding new data to the TimeZone formatter, I don't think we should implement this until someone asks us for it specifically.

This is not needed for ECMAScript, I believe.

**Comment by robertbastian** [comment]
We already have the data though (`IanaMapper`), so adding this as a date time field set using that should be straightforward. Most applications will already include this data anyway.

The limitation is that we will always produce the canonical IANA ID, not the input one.

**Comment by sffc** [comment]
We have the data available, but it isn't being linked into DateTimeFormatter, so this involves adding a bunch of APIs for loading the data and also increasing the stack size of the DateTimeFormatter. I would rather not merge such a change until there is a client need.
